lib/segment/src/common/operation_time_statistics.rs

```diff
@@ -368,9 +373,12 @@ fn merge_histograms(
     total_a: usize,
     total_b: usize,
 ) -> Vec<(f32, usize)> {
+    let len = itertools::merge(a.iter().map(|(le, _)| le), b.iter().map(|(le, _)| le))
```

Comment:
Since the lists must be sorted for `dedup`, shall we do an `debug_assert!()` or something here?

Reply:
This may be better: just take the sum of the two and allocate that. On the end we can do `shrink_to_fit` if desired.

It's not very essential for this to have an accurate count.

Reply:
These should be sorted anyway for this function to work regardless of estimating the capacity.